lib/msal-common/src/client/OnBehalfOfClient.ts

```diff
@@ -88,12 +89,16 @@ export class OnBehalfOfClient extends BaseClient {
 
         const credentialCache: CredentialCache = this.cacheManager.getCredentialsFilteredBy(accessTokenFilter);
         const accessTokens = Object.keys(credentialCache.accessTokens).map(key => credentialCache.accessTokens[key]);
-        if (accessTokens.length < 1) {
+
+        const numAccessTokens = accessTokens.length;
+        if (numAccessTokens < 1) {
             return null;
+        } else if (numAccessTokens > 1) {
+            throw ClientAuthError.createMultipleMatchingTokensInCacheError();
         }
         return accessTokens[0] as AccessTokenEntity;
     }
-    
+
```

Comment:
Should this be done in the silent flow as well?

Reply:
Yes. This logic is no longer in SilentFlowClient, it was refactored into CacheManager. I updated CacheManager as part of this PR.